internal/stack/aggregate.go

```diff
@@ -0,0 +1,55 @@
+package stack
+
+import (
+	"sort"
+
+	"github.com/buildpacks/pack/internal/dist"
+	"github.com/buildpacks/pack/internal/stringset"
+)
+
+// Aggregate aggregates two set of stacks into a merged set of stacks where compatibility exists.
+//
+// Examples:
+//
+// 	stacksA = [{ID: "stack1", mixins: ["build:mixinA", "mixinB", "run:mixinC"]}}]
+// 	stacksB = [{ID: "stack1", mixins: ["build:mixinA", "run:mixinC"]}}]
+// 	result = [{ID: "stack1", mixins: ["build:mixinA", "mixinB", "run:mixinC"]}}]
+//
+// 	stacksA = [{ID: "stack1", mixins: ["build:mixinA"]}}, {ID: "stack2", mixins: ["mixinA"]}}]
+// 	stacksB = [{ID: "stack1", mixins: ["run:mixinC"]}}, {ID: "stack2", mixins: ["mixinA"]}}]
+// 	result = [{ID: "stack1", mixins: ["build:mixinA", "run:mixinC"]}}, {ID: "stack2", mixins: ["mixinA"]}}]
+//
+// 	stacksA = [{ID: "stack1", mixins: ["build:mixinA"]}}, {ID: "stack2", mixins: ["mixinA"]}}]
+// 	stacksB = [{ID: "stack2", mixins: ["mixinA", "run:mixinB"]}}]
+// 	result = [{ID: "stack2", mixins: ["mixinA", "run:mixinB"]}}]
+//
+// 	stacksA = [{ID: "stack1", mixins: ["build:mixinA"]}}]
+// 	stacksB = [{ID: "stack2", mixins: ["mixinA", "run:mixinB"]}}]
+// 	result = []
+//
+func Aggregate(stacksA []dist.Stack, stacksB []dist.Stack) []dist.Stack {
+	set := map[string][]string{}
+
+	for _, s := range stacksA {
+		set[s.ID] = s.Mixins
```

Comment:
This wouldn't allow two stack entries with the same ID in `stacksA`. For instance:

```
stacksA = [
  {ID: "stack1", mixins: ["build:mixinA", "mixinB", "run:mixinC"]}},
  {ID: "stack1", mixins: ["build:mixinX", "mixinY", "run:mixinZ"]}}
]
```

The above is legal, right @sclevine?

Reply:
It isn't legal for a buildpack to specify the same stack ID twice with multiple sets of mixins. So I think this is okay.

Reply:
OK, sounds good!